src/Compilers/Core/MSBuildTask/Microsoft.CSharp.Core.targets

```diff
@@ -48,6 +48,12 @@
       <PdbFile Condition="'$(PdbFile)' == '' AND '$(OutputType)' == 'winmdobj' AND '$(_DebugSymbolsProduced)' == 'true'">$(IntermediateOutputPath)$(TargetName).compile.pdb</PdbFile>
     </PropertyGroup>
 
+    <PropertyGroup>
+      <LangVersion Condition="'$(LangVersion)' == '' AND
+        (('$(TargetFrameworkIdentifier)' == '.NETCore' AND '$(TargetFrameworkVersion)' == 'v3.0') OR
+         ('$(TargetFrameworkIdentifier)' == '.NETStandard' OR '$(TargetFrameworkVersion)' == 'v2.1'))">preview</LangVersion>
+    </PropertyGroup>
```

Comment:
There's no case where you could install this into a non-preview SDK? like if somebody requested an older version or the like?

Seems like there should be a property "IsPreview" we should be able to depend on...

Reply:
We're going to remove this before shipping

Reply:
Before shipping which version?  If I understood correctly, the issue @jasonmalinowski raises would be forcing the compiler via compiler nupkg the 16.0 compiler into a netcoreapp3.0 project using 3.0 RTM SDK. That would be a very strange thing to do, but I suppose the behavior would be incorrect.

Reply:
As long as that's "strange", it's fine with me. :smile:

Reply:
"Strange" there is just my gut opinion. I don't know how often folks downgrade the compiler like that.

Reply:
That would indeed be strange. I'm not really worried about that use case (where someone using a 3.0 RTM SDK uses NuGet to downgrade to the 16.0 RTM compiler and gets stuck with a silent preview version of C# 8). I think our guidance should be that if you're using a compiler from NuGet you should also explicitly set your language version. Does that sound right @jaredpar ?

Reply:
I'm not concerned with that case.

Is there a reason we didn't write it by conditioning the entire property group like: 

``` xml
    <PropertyGroup Condition="'$(LangVersion)' == '' 
      <LangVersion Condition="'$(TargetFrameworkIdentifier)' == '.NETCoreApp' AND '$(TargetFrameworkVersion)' == 'v3.0')"preview</LangVersion>
      <LangVersion Condition="'$(TargetFrameworkIdentifier)' == '.NETStandard' AND '$(TargetFrameworkVersion)' == 'v2.1')"preview</LangVersion>
    </PropertyGroup>
```

This way seems more idiomatic. Or is that just my warped perception?

Reply:
cc @rainersigwald 

I'm not sure what is "idiomatic", but I find what @jaredpar wrote to be easier to follow (subjectively).

Reply:
@agocke solution obviously works. I just don't have a feel for what is better, more idiomatic here. Don't want to block the PR on this though. Can clean up later if we decide there is a better way to do this that we must do.

Reply:
Since I went through a fair bit of manual testing for the existing change I'd prefer we ship as-is and clean up later.

Reply:
FBM. Don't treat me comment as blocking. Curious mostly.